aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -42,16 +42,22 @@ pub(crate) fn get_resource_group_from_metadata(
 /// Adapter to convert a `StateView` into a `MoveResolverExt`.
 pub struct StorageAdapter<'a, S> {
     state_store: &'a S,
+    accurate_byte_count: bool,
     resource_group_cache:
         RefCell<BTreeMap<AccountAddress, BTreeMap<StructTag, BTreeMap<StructTag, Vec<u8>>>>>,
 }
 
 impl<'a, S: StateView> StorageAdapter<'a, S> {
     pub fn new(state_store: &'a S) -> Self {
-        Self {
+        let mut s = Self {
             state_store,
+            accurate_byte_count: false,
             resource_group_cache: RefCell::new(BTreeMap::new()),
-        }
+        };
+        if gas_config(&s).1 >= 9 {
```

Comment:
you need to set the version to 9 in this PR too otherwise during upgrade it won't be set.